parser/pbf/pbf.go

```diff
@@ -30,20 +34,47 @@ func readDenseNodes(
 	coordScale := 0.000000001
 	lastKeyValPos := 0
 
+	var lastDenseinfoTimestamp int64
+	var lastDenseinfoChangeset int64
+	var lastDenseinfoUid int32
+	var lastDenseinfoUserSid int32
+
 	for i := range coords {
 		lastId += dense.Id[i]
 		lastLon += dense.Lon[i]
 		lastLat += dense.Lat[i]
+
+		lastDenseinfoTimestamp += dense.Denseinfo.Timestamp[i]
+		lastDenseinfoChangeset += dense.Denseinfo.Changeset[i]
+		lastDenseinfoUid += dense.Denseinfo.Uid[i]
+		lastDenseinfoUserSid += dense.Denseinfo.UserSid[i]
+
 		coords[i].Id = lastId
 		coords[i].Long = (coordScale * float64(lonOffset+(granularity*lastLon)))
 		coords[i].Lat = (coordScale * float64(latOffset+(granularity*lastLat)))
+
+		DenseinfoVersion := dense.Denseinfo.Version[i]
+
+		DenseinfoTimestamp := time.Unix(lastDenseinfoTimestamp, 0)
+		DenseinfoChangeset := lastDenseinfoChangeset
+		DenseinfoUid := lastDenseinfoUid
+		DenseinfoUserSid := lastDenseinfoUserSid
+
 		if stringtable != nil && len(dense.KeysVals) > 0 {
 			if dense.KeysVals[lastKeyValPos] != 0 {
 				tags := parseDenseNodeTags(stringtable, &dense.KeysVals, &lastKeyValPos)
+
 				if tags != nil {
+
 					if _, ok := tags["created_by"]; ok && len(tags) == 1 {
 						// don't add nodes with only created_by tag to nodes cache
 					} else {
+						tags["osm_version"] = strconv.FormatInt(int64(DenseinfoVersion), 10)
```

Comment:
Done,

I have creted some bool "const" for generating optimal code,  
and  I have moved them to "./config/config.go" - files.

---- see:
// add special parsed meta tags to key list ( version , user, ... )
const ParseMetadata = true
const ParseMetadataPrefix = "osm_"

const ParseMetadataVarVersion = true
const ParseMetadataVarTimestamp = true
const ParseMetadataVarChangeset = true
const ParseMetadataVarUid = true
const ParseMetadataVarUser = true